### ISSUE TITLE
Remaining things to improve/correct in mock_data

### ISSUE THREAD

**Issue by rherbonnet** [LSSTDESC/CLMM#312]
- [x] In mock_data._find_aphysical_galaxies there's the lines:
`    badgals = np.where((np.abs(galaxy_catalog['e1']) > 1.0) |
                       (np.abs(galaxy_catalog['e2']) > 1.0) |
                       (galaxy_catalog['ztrue'] < zsrc_min)
                      )[0] `
So all galaxies with either |e1|>1 or |e2|>1 are rejected. But shouldn't it be rejecting |e|=sqrt(e1 x e1+e2 x e2)>1, instead?

- [x] In `_generate_galaxy_catalog`, there remain some hard-coded `halo_profile_model='nfw'` that should be changed to `halo_profile_model=halo_profile_model` as this is a user-defined keyword. With the Numcosmo backend, other profiles are available.

Once #311 is merged:

- [x] Suggestion to add another keyword `massdef` that can be equal to `mean`, `critical` or `virial`. At the moment, mock_data can only operate with the mean overdensity mass definition. With the CCL and NC backend, the user could work with the other definitions and the option should be available in mock_data.  

- [x] Also, Delta_SO is a non-optional argument of `generate_galaxy_catalog`. Suggest making it an optional keyword, that defaults to 200, for consistency with `massdef` being a optional keyword as well.

**Comment by combet** [comment]
@rherbonnet, I hijacked your issue to add 2 more things that we should take care of in mock_data. Didn't want to open a separate issue as they all concern the same file.

**Comment by shenmingfu** [comment]
In Example 1, binning would cause fitted mass larger than input. The reason is that g(r) is a convex function -- the function value after binning would be larger, but the bias becomes smaller as bin number increases. I will try to add some description to Example1 to highlight this.
